hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -207,11 +207,30 @@ public static DataStream<Object> append(
       Configuration conf,
       RowType rowType,
       DataStream<RowData> dataStream) {
-    WriteOperatorFactory<RowData> operatorFactory = AppendWriteOperator.getFactory(conf, rowType);
+    boolean isBucketIndex = OptionsResolver.isBucketIndexType(conf);
+    if (isBucketIndex) {
```

Comment:
Why do we need bucket index in append mode?

Reply:
Two reasons come to my mind. First of all, we support simple bucket index for bulk insert write into Hudi table using Flink. The second one is possibility at least skip some data during reading by bucket ID if it's applicable.